testing/file_test/autoupdate.cpp

```diff
@@ -294,7 +295,7 @@ auto FileTestAutoupdater::StartSplitFile() -> void {
     AddCheckLines(stdout_, /*to_file_end=*/false);
   }
 
-  ++non_check_line_;
```

Comment:
This doesn't really seem like an improvement to me... Could we disable this one too?

Reply:
To be sure you have context, this is [bugprone-pointer-arithmetic-on-polymorphic-object](https://clang.llvm.org/extra/clang-tidy/checks/bugprone/pointer-arithmetic-on-polymorphic-object.html). I'm hesitant to disable the bugprone warning due to what it's looking for, are you sure we don't want it?

Reply:
Thought of a better fix here, with `final`

Reply:
Oh, I see. It's about the polymorphic class!

Yeah, I think `final` is better here. I think `std::advance` would be just as bug prone as the `++`. But the `final` is a pretty nice improvement.